x-pack/plugins/infra/server/plugin.ts

```diff
@@ -4,30 +4,134 @@
+import { METRICS_FEATURE, LOGS_FEATURE } from './features';
+import { UsageCollector } from './usage/usage_collector';
+import { InfraStaticSourceConfiguration } from './lib/sources/types';
 
-export class InfraPlugin implements Plugin<InfraSetup> {
-  private readonly initContext: PluginInitializerContext;
+export const config = {
+  schema: schema.object({
+    enabled: schema.maybe(schema.boolean()),
+    query: schema.object({
+      partitionSize: schema.maybe(schema.number()),
+      partitionFactor: schema.maybe(schema.number()),
+    }),
+  }),
+};
 
-  constructor(initContext: PluginInitializerContext) {
-    this.initContext = initContext;
+export type InfraConfig = TypeOf<typeof config.schema>;
+
+export interface KbnServer extends Server {
+  usage: any;
+}
+
+export interface InfraPluginSetup {
+  defineInternalSourceConfiguration: (
+    sourceId: string,
+    sourceProperties: InfraStaticSourceConfiguration
+  ) => void;
+}
+
+const DEFAULT_CONFIG: InfraConfig = {
+  enabled: true,
+  query: {
+    partitionSize: 75,
+    partitionFactor: 1.2,
+  },
+};
+
+export class InfraServerPlugin {
+  public config: InfraConfig = DEFAULT_CONFIG;
+  public libs: InfraBackendLibs | undefined;
+
+  constructor(context: PluginInitializerContext) {
+    const config$ = context.config.create<InfraConfig>();
+    config$.subscribe(configValue => {
+      this.config = {
+        ...DEFAULT_CONFIG,
+        enabled: configValue.enabled,
+        query: {
+          ...DEFAULT_CONFIG.query,
+          ...configValue.query,
+        },
+      };
+    });
   }
 
-  public setup() {
+  getLibs() {
+    if (!this.libs) {
+      throw new Error('libs not set up yet');
+    }
+    return this.libs;
+  }
+
+  setup(core: CoreSetup, plugins: InfraServerPluginDeps) {
+    const framework = new KibanaFramework(core, this.config, plugins);
+    const sources = new InfraSources({
+      config: this.config,
+    });
+    const sourceStatus = new InfraSourceStatus(
+      new InfraElasticsearchSourceStatusAdapter(framework),
+      {
+        sources,
+      }
+    );
+    const snapshot = new InfraSnapshot({ sources, framework });
+    const logAnalysis = new InfraLogAnalysis({ framework });
+
+    // TODO: separate these out individually and do away with "domains" as a temporary group
+    const domainLibs: InfraDomainLibs = {
+      fields: new InfraFieldsDomain(new FrameworkFieldsAdapter(framework), {
+        sources,
+      }),
+      logEntries: new InfraLogEntriesDomain(new InfraKibanaLogEntriesAdapter(framework), {
+        sources,
+      }),
+      metrics: new InfraMetricsDomain(new KibanaMetricsAdapter(framework)),
+    };
+
+    this.libs = {
+      configuration: this.config,
+      framework,
+      logAnalysis,
+      snapshot,
+      sources,
+      sourceStatus,
+      ...domainLibs,
+    };
+
+    plugins.features.registerFeature(METRICS_FEATURE);
+    plugins.features.registerFeature(LOGS_FEATURE);
+
```

Comment:
The call to the `home` plugin doesn't exist here as it does within the [`new_platform_plugin.ts`](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/infra/server/new_platform_plugin.ts#L116) file, e.g.:

```
plugins.home.sampleData.addAppLinksToSampleDataset('logs', [
      {
        path: `/app/${APP_ID}#/logs`,
        label: logsSampleDataLinkLabel,
        icon: 'logsApp',
      },
    ]);
```